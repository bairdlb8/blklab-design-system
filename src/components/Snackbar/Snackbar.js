import React, { useEffect, useState } from 'react'
import PropTypes from 'prop-types'

import './Snackbar.scss'

const Snackbar = ({ className = '', children, status = 'idle', onDismiss }) => {
  const [state, setState] = useState({
    uiState: 'hidden'
  })

  useEffect(() => {
    setTimeout(() => {
      setState({ ...state, uiState: 'show' })
    }, 100)
  }, [])

  return (
    <div className={`ds-snackbar ${state.uiState} ${status}`}>
      <div className="ds-snackbar__message">{children}</div>
      {onDismiss && (
        <div className="ds-snackbar__action" onClick={onDismiss}>
          Dismiss
        </div>
      )}
    </div>
  )
}

Snackbar.propTypes = {}

export default Snackbar
