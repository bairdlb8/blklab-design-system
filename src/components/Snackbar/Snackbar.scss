.ds-snackbar {
  --snackbar-color: #fff;
  padding: var(--spacer-2) var(--spacer-4);
  background-color: var(--info-color);
  transition: all var(--animation-speed-fast) linear;
  display: grid;
  grid-template-columns: 4fr max-content;
  opacity: 0;
  transform-origin: center;
  transform: scale(0.98) translateY(10px);
  transition-delay: 0.25s;
  transition: all var(--animation-speed-fast) var(--animation-easing-circ-out);

  &.show {
    opacity: 1;
    transform: scale(1) translateY(0);
  }

  &.error {
    background-color: var(--error-color);
  }

  &.warning {
    background-color: var(--warning-color);
  }

  &.success {
    background-color: var(--success-color);
  }

  .floating {
    position: absolute;
  }

  &__message {
    color: var(--snackbar-color);
  }

  &__action {
    color: var(--snackbar-color);
    cursor: pointer;
    align-items: flex-end;
  }
}
