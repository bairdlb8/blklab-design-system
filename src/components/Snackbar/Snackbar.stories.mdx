import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks'

import Snackbar from './Snackbar'

<Meta title="Components / Snackbar" component={ Snackbar } />

# Snackbar

## Add some text here

export const Template = (args) => {
  return <Snackbar {...args} />
}

<Preview withToolbar>
  <Story name="Primary" args={ {} }>
    {Template.bind({})}
  </Story>
</Preview>

<Props of={ Snackbar } />
