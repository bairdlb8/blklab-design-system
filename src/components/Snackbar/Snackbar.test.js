import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { axe, toHaveNoViolations } from 'jest-axe'
import Snackbar from './Snackbar'

expect.extend(toHaveNoViolations)

describe('<Snackbar />', () => {
  describe('accessibility', () => {
    it('should not fail any accessibility tests', async () => {
      const { container } = render(<Snackbar />)
      expect(await axe(container)).toHaveNoViolations()
    })
  })

  it('should take a snapshot', () => {
    const { asFragment } = render(<Snackbar />)

    expect(asFragment(<Snackbar />)).toMatchSnapshot()
  })
})
