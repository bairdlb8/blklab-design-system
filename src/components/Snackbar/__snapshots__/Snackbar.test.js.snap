// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Snackbar /> should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    Snackbar
  </div>
</DocumentFragment>
`;
